### ISSUE TITLE
execute make error

### ISSUE THREAD

**Issue by hellopower** [greenplum-db/gpos#33]
``` shell
[root@BJYF-Thallo-145135 build]# make
[  1%] Building CXX object CMakeFiles/gpos.dir/libgpos/src/common/CStackDescriptor.cpp.obj
In file included from /data0/wubiao/package/HAWQ/depends/thirdparty/gpos/libgpos/include/gpos/types.h:28:0,
                 from /data0/wubiao/package/HAWQ/depends/thirdparty/gpos/libgpos/include/gpos/utils.h:18,
                 from /data0/wubiao/package/HAWQ/depends/thirdparty/gpos/libgpos/src/common/CStackDescriptor.cpp:12:
/data0/wubiao/package/HAWQ/depends/thirdparty/gpos/libgpos/src/common/CStackDescriptor.cpp: In member function 'void gpos::CStackDescriptor::BackTrace(gpos::ULONG)':
/data0/wubiao/package/HAWQ/depends/thirdparty/gpos/libgpos/include/gpos/assert.h:46:67: error: size of array 'assert_array' is negative
 #define GPOS_CPL_ASSERT(x)  extern int assert_array[ (x) ? 1 : -1 ]
                                                                   ^
/data0/wubiao/package/HAWQ/depends/thirdparty/gpos/libgpos/src/common/CStackDescriptor.cpp:167:2: note: in expansion of macro 'GPOS_CPL_ASSERT'
  GPOS_CPL_ASSERT(!"Backtrace is not supported for this platform");
  ^
/data0/wubiao/package/HAWQ/depends/thirdparty/gpos/libgpos/include/gpos/assert.h:46:40: error: unused variable 'gpos::assert_array' [-Werror=unused-variable]
 #define GPOS_CPL_ASSERT(x)  extern int assert_array[ (x) ? 1 : -1 ]
                                        ^
/data0/wubiao/package/HAWQ/depends/thirdparty/gpos/libgpos/src/common/CStackDescriptor.cpp:167:2: note: in expansion of macro 'GPOS_CPL_ASSERT'
  GPOS_CPL_ASSERT(!"Backtrace is not supported for this platform");
  ^
At global scope:
cc1plus: error: unrecognized command line option "-Wno-tautological-undefined-compare" [-Werror]
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/gpos.dir/libgpos/src/common/CStackDescriptor.cpp.obj] Error 1
make[1]: *** [CMakeFiles/gpos.dir/all] Error 2
make: *** [all] Error 2

```
Thanks

**Comment by danielgustafsson** [comment]
Which compiler version are you using?

**Comment by hellopower** [comment]
@danielgustafsson 
``` shell
[hadoop@BJYF-Thallo-145135 HAWQ]$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/local/libexec/gcc/x86_64-unknown-linux-gnu/4.8.0/lto-wrapper
Target: x86_64-unknown-linux-gnu
Configured with: ../gcc-4.8.0/configure --enable-checking=release --enable-languages=c,c++ --disable-multilib
Thread model: posix
gcc version 4.8.0 (GCC)
```
Thanks

**Comment by VinnyPembe** [comment]
Hello, was there any fix for this issue at all?

I'm getting almost the exact same error as above:

> [root@mdw build]# make
[  0%] Building CXX object libgpos/CMakeFiles/gpos.dir/src/common/CStackDescriptor.cpp.o
In file included from /usr/local/gpdb/include/gpos/types.h:28:0,
                 from /usr/local/gpdb/include/gpos/utils.h:18,
                 from /root/gporca/libgpos/src/common/CStackDescriptor.cpp:12:
/root/gporca/libgpos/src/common/CStackDescriptor.cpp: In member function ‘void gpos::CStackDescriptor::BackTrace(gpos::ULONG)’:
/usr/local/gpdb/include/gpos/assert.h:46:67: error: size of array ‘assert_array’ is negative
 #define GPOS_CPL_ASSERT(x)  extern int assert_array[ (x) ? 1 : -1 ]
                                                                   ^
/root/gporca/libgpos/src/common/CStackDescriptor.cpp:167:2: note: in expansion of macro ‘GPOS_CPL_ASSERT’
  GPOS_CPL_ASSERT(!"Backtrace is not supported for this platform");
  ^
/usr/local/gpdb/include/gpos/assert.h:46:40: error: unused variable ‘gpos::assert_array’ [-Werror=unused-variable]
 #define GPOS_CPL_ASSERT(x)  extern int assert_array[ (x) ? 1 : -1 ]
                                        ^
/root/gporca/libgpos/src/common/CStackDescriptor.cpp:167:2: note: in expansion of macro ‘GPOS_CPL_ASSERT’
  GPOS_CPL_ASSERT(!"Backtrace is not supported for this platform");
  ^
At global scope:
cc1plus: error: unrecognized command line option "-Wno-tautological-undefined-compare" [-Werror]
cc1plus: all warnings being treated as errors
make[2]: *** [libgpos/CMakeFiles/gpos.dir/src/common/CStackDescriptor.cpp.o] Error 1
make[1]: *** [libgpos/CMakeFiles/gpos.dir/all] Error 2
make: *** [all] Error 2
[root@mdw build]# 
[root@mdw build]# 
[root@mdw build]# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/4.8.5/lto-wrapper
Target: x86_64-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-linker-hash-style=gnu --enable-languages=c,c++,objc,obj-c++,java,fortran,ada,go,lto --enable-plugin --enable-initfini-array --disable-libgcj --with-isl=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/isl-install --with-cloog=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/cloog-install --enable-gnu-indirect-function --with-tune=generic --with-arch_32=x86-64 --build=x86_64-redhat-linux
Thread model: posix
gcc version 4.8.5 20150623 (Red Hat 4.8.5-11) (GCC) 
[root@mdw build]# 
[root@mdw build]# 
[root@mdw build]# uname -r
4.9.7-1.el7.elrepo.x86_64

Cheers!

**Comment by danielgustafsson** [comment]
I had forgotten about this issue, sorry about that =(. GPOS is now rolled into GPORCA rather than being a stand-alone library, are you getting this error as part of building GPORCA?

**Comment by VinnyPembe** [comment]
Yes, I am. I actually installed GP_Xerces, then GPOS (stand-alone), and now onto attempting to install GPOrca. Could having installed GPOS first have caused the problem?

I can very easily re-build the VM from scratch and try to install it without GPOS first if that may be the issue.

**Comment by PengJi** [comment]
Remove the cmake files generated under build folder of gporca repo:
`rm -fr build/*`
Remove gporca header files and library, (assuming the default install prefix /usr/local)
`rm -rf /usr/local/include/naucratesm  
rm -rf /usr/local/include/gpdbcost
rm -rf /usr/local/include/gpopt
rm -rf /usr/local/include/gpos
rm -rf /usr/local/lib/libnaucrates.so*
rm -rf /usr/local/lib/libgpdbcost.so*
rm -rf /usr/local/lib/libgpopt.so*
rm -rf /usr/local/lib/libgpos.so*`
